src/status_im2/common/bottom_sheet_screen/view.cljs

```diff
@@ -46,40 +43,54 @@
 
 (defn- f-view
   [_]
-  (let [scroll-enabled (reagent/atom true)
-        curr-scroll    (reagent/atom 0)]
+  (let [scroll-enabled      (reagent/atom true)
+        curr-scroll         (reagent/atom 0)
+        animating?          (reagent/atom true)
```

Comment:
An atom that says if the current sheet is being animated or not.
I took the solution suggested by @smohamedjavid  in a previous PR (https://github.com/status-im/status-mobile/pull/17891#pullrequestreview-1741734622)

Reply:
Why reagent atoms? Why not shared value?

Reply:
@Parveshdhull I don't think these atoms are used in animations

Reply:
Still, the use of reagent atoms looks scary. Before you know, you are passing the same reagent atoms across several files (as we did in the messages screen), and a small reset force updates a whole bunch of screens.

They are not animations, still sharedValue value can also update the required parameters like reagent atoms without force updating the view. In other words

1. animating? - animates property window-size
2. scroll-enabled - animates property scroll-enabled
3. curr-scroll - It looks like it just holds data that is used in drag-gesture, we can just use a simple atom here.

Reply:
Hey @Parveshdhull 

> ... a small reset force updates a whole bunch of screens.

Hmm... it doesn't work like that, as I said in this PR, I'm using a late deref approach, this aproach is also being used for the other atoms in this component. 

Regarding to:

> Before you know, you are passing the same reagent atoms across several files (as we did in the messages screen)

Probably the issue with messages screns was due to a different performance topic.

Remember Reagent does some optimizations before sending things to react, and if we don't perform a `deref`, the view is not re-rendered. Here's a small working example of the late deref approach and rerendering, that you can copy and test it:

```Clojure
(defn subcomponent-2 [{:keys [a-2]}]
  (println "subcomponent-2 repainted")
  ;; `a-2` is deref'ed only here, but traveled through all components.
  [quo/text (str "atom 2 value:" @a-2)])

(defn subcomponent-1 [{:keys [a-1 a-2]}]
  (println "subcomponent-1 repainted")
  [rn/view
   [quo/text (str "atom 1 value:" @a-1)]  ;; Here only `a-1` uses deref
   [subcomponent-2 {:a-2 a-2}]]) ;; Here we pass the atom `a-2`

(defn test-component []
  (let [atom-1        (reagent/atom 0)
        atom-2        (reagent/atom 0)
        update-atom-1 #(swap! atom-1 inc)
        update-atom-2 #(swap! atom-2 inc)]
    (fn []
      (println "test-component repainted")
      [rn/view
      ;; Atoms are being passed, we'll do the deref only inside the subcomponents (as in this bottom sheet component)
       [subcomponent-1 {:a-1 atom-1
                        :a-2 atom-2}]
       [quo/button {:on-press update-atom-1}
        "Update atom 1"]
       [quo/button {:on-press update-atom-2}
        "Update atom 2"]])))
```

It'll render the following:

![image](https://github.com/status-im/status-mobile/assets/90291778/055a7ac8-7c7f-4a37-bfa4-64e6351543c2)

Here is the result in the terminal after pressing the `update atom 2` button 3 times, and after that pressing the `update atom 1` three times:

```
  subcomponent-2 repainted
  subcomponent-2 repainted
  subcomponent-2 repainted
  
  subcomponent-1 repainted
  subcomponent-1 repainted
  subcomponent-1 repainted
```

So, it's easy to see that `reset`ing (or `swap`ing) the atom-2 didn't re-render the intermediate components.

On the other hand, if we pass `deref`ed values instead of atoms:
```clojure
;; Changing this call inside `test-component` and also removing the `@` insde subcomponents
[subcomponent-1 {:a-1 @atom-1
                 :a-2 @atom-2}]
```

then when press the `update atom 2` button, the terminal will print:
```
test-component repainted
subcomponent-1 repainted
subcomponent-2 repainted
```

In that case it's actually re executing the render functions of the intermediate components.
But, as said before, that's not the approach in this component.

So @Parveshdhull wdyt?
IMO we shouldn't use shared values for something that aren't animations. Seems like a misuse.

Reply:
Thank you @ulisesmac for detailed explanation. Using sharedValue here, don't make much sense.
Should we just use simple atom for `curr-scroll`, it looks like it is just storing data?

Reply:
Also, probably not in this PR, but should we consider using worklets for drag gesture for further improving performance?

Reply:
Sorry @Parveshdhull , I forgot to reply to it.

> Should we just use simple atom for curr-scroll, it looks like it is just storing data?

I checked the rerenderings, and at least in the emoji picker it's not rerendering each time someone scrolls (since no one is performing `deref` on that atom).

I think changing that for a regular atom might create an unexpected behvaior in other screens since maybe some screens use it to trigger a re-render.

As said before, since this is not harmful in this screen, we can just keep it.

Regarding to:
> Also, probably not in this PR, but should we consider using worklets for drag gesture for further improving performance?
Yes,  maybe gestures can use worklets :) we could open an issue to improve the performance of the bottom sheet :+1: